velox/core/PlanNode.cpp

```diff
@@ -504,6 +504,18 @@ void addWindowFunction(
   if (frame.endType == WindowNode::BoundType::kUnboundedPreceding) {
     VELOX_USER_FAIL("Window frame end cannot be UNBOUNDED PRECEDING");
   }
+  if (frame.type == WindowNode::WindowType::kRange) {
+    if (frame.startType == WindowNode::BoundType::kPreceding ||
+        frame.startType == WindowNode::BoundType::kFollowing) {
+      VELOX_USER_FAIL(
```

Comment:
Should this be VELOX_NYI and not VELOX_USER_FAIL. We should implement kPreceding/kFollowing for RANGE mode in Velox.